### DISCUSSION TITLE
Compile ok but run error

### DISCUSSION THREAD

**Discussion by saltball** [deepmodeling/abacus-develop#2550] [Q&A]
I tried to compile ABACUS from source, and I used the latest OneAPI toolchain, but it always strangely compiles well then has numerical problems running. ( file `abacus` also appears to be working properly, but none of the results are reproducible tests in `tests/integrate`.)

I would like to know if the community has a complete test of the compilation toolchain?

BTW, I wonder if this is related to https://github.com/deepmodeling/abacus-develop/discussions/2299

**Comment by caic99** [comment]
Hi @saltball ,
Please submit an issue and provide details of your problems there.

**Comment by QuantumMisaka** [comment]
Hi @saltball 
There is true that the full test of ABACUS compiled by Toolchain is lack, but the toolchain only facilitates the compilation of dependence of ABACUS, it should have no difference in precision with ABACUS in docker or other env.  Your problem seems interesting, could you provide more information and submit an issue ?

**Comment by saltball** [comment]
@caic99 @QuantumMisaka Thank you for your attention. This discussion I encountered six months ago, where I was getting errors in the `tests/integrate` when using the compiled files from the source code. However, after pull the latest code, I no longer encountered significant result inconsistencies. At that time, I converted this discussion into an issue #2727  , but since I quickly realized that the problem no longer existed, I closed that issue. Therefore, I believe this discussion may no longer be necessary.

---

BYW, I am still interested to know if the developers have any reasonable expectations regarding the current `tests` and whether the included tests are expected to pass with sufficient accuracy. Or is the testing work still ongoing? Because today I noticed errors in some test cases when installing ABACUS via `conda`.

For example, I got `warning.log` in `101_PW_upf201_uspp_NaCl` and with follows:
```
 In SCAN_BEGIN, can't find: LATTICE_VECTORS block.
 In SCAN_BEGIN, can't find: <PP_Q> block.
 In SCAN_END, can't find: </PP_Q> block.
```
I'm not sure if it is trouble about my machines. I just run about 80 tests because it costs. While most other pervious tests are normal, except `101_PW_15_paw` warns me `Input  warning : to use PAW, compile with USE_PAW`. And few cases have errors about 1e-5 (I think it doesn't matter).

---

I would recommend providing clarification on whether these tests are expected to pass or if they are expected to pass under `certain compilation conditions, architectures, or dependencies`. It may be necessary to conduct tests in different environments, and I am willing to assist in that regard. However, it is important for me to know whether these tests are all expected to pass.
